@xen-orchestra/proxy/src/app/mixins/backups/_RemoteAdapter.js

```diff
@@ -371,23 +371,45 @@ export class RemoteAdapter {
     const backups = []
 
     try {
-      const files = await handler.list(`${BACKUP_DIR}/${vmUuid}`, {
-        filter: isMetadataFile,
-        prependDir: true,
-      })
-      await asyncMap(files, async file => {
+      const files = (
+        await handler.list(`${BACKUP_DIR}/${vmUuid}`, {
+          filter: isMetadataFile,
+          prependDir: true,
+        })
+      ).sort()
+
+      const chainSizeByVdi = {}
+      for (const file of files) {
         try {
           const metadata = await this.readVmBackupMetadata(file)
           if (predicate === undefined || predicate(metadata)) {
             // inject an id usable by importVmBackupNg()
             metadata.id = metadata._filename
 
+            // a backup size is a sum of all its VHDs' size
+            // a VHD size is a sum of the VHDs' chain size
+            if (metadata.mode === 'delta') {
```

Comment:
We need to handle full backups as well.

Reply:
The full backups' metadata contain the property size.